packages/issuer/lib/VcIssuer.ts

```diff
@@ -0,0 +1,49 @@
+import { CredentialIssuerCallback, CredentialRequest, CredentialResponse, IssuerMetadata, TokenErrorResponse } from '@sphereon/openid4vci-common'
+import { ICredential, W3CVerifiableCredential } from '@sphereon/ssi-types'
+
+export class VcIssuer {
+  _issuerMetadata: IssuerMetadata
+  _userPinRequired?: boolean
+  _issuerCallback?: CredentialIssuerCallback
+
+  constructor(issuerMetadata: IssuerMetadata, userPinRequired?: boolean) {
+    this._issuerMetadata = issuerMetadata
+    this._userPinRequired = userPinRequired
+  }
+
+  public getIssuerMetadata() {
+    return this._issuerMetadata
+  }
+
+  public async issueCredentialFromIssueRequest(issueCredentialRequest: CredentialRequest): Promise<CredentialResponse> {
+    //TODO: do we want additional validations here?
+    if (this.isMetadataSupportCredentialRequestFormat(issueCredentialRequest.format)) {
+      return {
+        credential: await this.issueCredential({ credentialRequest: issueCredentialRequest }),
```

Comment:
Probably wise to have an issuerCallback as optional argument for this method. If supplied that one is used instead of the callback set at the class level. Sometimes you need more specific context/variables when issuing a credential. Also do not only supply the arguments to the callback function alone. The whole callback function would need to be passed in next to the arguments, so an implementer has full control over how they want to issue

Reply:
added optional parameter for callback function here